src/blob_gc_job.cc

```diff
@@ -184,12 +184,12 @@ Status BlobGCJob::DoSample(const BlobFileMeta* file, bool* selected) {
   if (*selected) return Status::OK();
 
   // TODO: add do sample count metrics
-  auto records_size = file->file_size() - BlobFileHeader::kEncodedLength -
+  auto records_size = file->file_size() - BlobFileHeader::kMaxEncodedLength -
```

Comment:
You need to add a `header_size()` method to `FileMetaData` and return result depend on header version. 

Actually leave an inline comment here to note `records_size` is not accurate is also fine. The `DoSample` method is being removed soon.